### PR TITLE
Reduce usage of InternalThreadLocalMap

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We should not use InternalThreadLocalMap whenever possible and just use FastThreadLocal.

Modifications:

Replace usage of InternalThreadLocalMap with FastThreadLocal when possible

Result:

Cleanup

### CODE REVIEW

**Comment by normanmaurer** [comment]
@chrisvest This is the first step... Once this is merged I will continue to with the hopefully outcome that `InternalThreadLocallMap` is not public anymore.

**Comment by normanmaurer** [comment]
@chrisvest I just did add the second commit here... Now `InternalThreadLocalMap` his completely hidden from the user.

**Comment by normanmaurer** [comment]
This also fixes Fixes #11929.

**Review by chrisvest — APPROVED** [review]

File: common/src/main/java/io/netty/util/concurrent/FastThreadLocalThread.java [hunk 1]
```diff
@@ -70,15 +69,15 @@ public FastThreadLocalThread(ThreadGroup group, Runnable target, String name, lo
      * Returns the internal data structure that keeps the thread-local variables bound to this thread.
      * Note that this method is for internal use only, and thus is subject to change at any time.
      */
-    public final InternalThreadLocalMap threadLocalMap() {
```
Comment (laosijikaichele) [inline common/src/main/java/io/netty/util/concurrent/FastThreadLocalThread.java:73]: @normanmaurer , IMHO, May be we use 'protected' access level might be better? That can make subclass able to access this method too.  Same below.

**Pull request merged**